### PR TITLE
MXCrypto: Encrypt the messages for invited members according to the h…

### PR DESCRIPTION
**Pull request by giomfo**
…istory visibility (if the option is enabled in MXCryptoConfig).

Related to PR #292

### CODE REVIEW

File: matrix-sdk/src/main/java/org/matrix/androidsdk/crypto/MXCrypto.java [hunk 1]
```diff
@@ -825,21 +825,6 @@ public void setDeviceVerification(final int verificationStatus, final String dev
...
-        for (Room room : rooms) {
-            if (room.isEncrypted()) {
-                RoomMember roomMember = room.getMember(userId);
-
-                // test if the user joins the room
-                if ((null != roomMember) && TextUtils.equals(roomMember.membership, RoomMember.MEMBERSHIP_JOIN)) {
-                    userRoomIds.add(room.getRoomId());
-                }
-            }
-        }
```
Comment (bmarty) [inline matrix-sdk/src/main/java/org/matrix/androidsdk/crypto/MXCrypto.java]: Ok to remove this dead code. Missing cleanup from 444644570ac6b424b700d38879a839247ba80f4f

File: matrix-sdk/src/main/java/org/matrix/androidsdk/crypto/MXCrypto.java [hunk 2]
```diff
@@ -2427,22 +2428,6 @@ public void onSuccess(MXUsersDevicesMap<MXDeviceInfo> devicesMap) {
...
-
```
Comment (bmarty) [inline matrix-sdk/src/main/java/org/matrix/androidsdk/crypto/MXCrypto.java]: Ok to remove this dead code. Missing cleanup from 444644570ac6b424b700d38879a839247ba80f4f

File: matrix-sdk/src/main/java/org/matrix/androidsdk/data/Room.java [hunk 3]
```diff
@@ -183,6 +183,16 @@ public IMXStore getStore() {
         return mStore;
     }
 
+    /**
+     * Determine whether we should encrypt messages for invited users in this room.
+     *
+     * @return true if we should encrypt messages for invited users.
+     */
+    public boolean shouldEncryptForInvitedMembers() {
```
Comment (bmarty) [inline matrix-sdk/src/main/java/org/matrix/androidsdk/data/Room.java]: From the Room class point of view, this method is not related to encryption.
I would rename it to `isHistoryVisibleForNotJoinedMembers` or something similar
Please also update the Javadoc accordingly

File: matrix-sdk/src/main/java/org/matrix/androidsdk/data/Room.java [hunk 4]
```diff
@@ -183,6 +183,16 @@ public IMXStore getStore() {
...
+        String historyVisibility = getLiveState().history_visibility;
```
Comment (bmarty) [inline matrix-sdk/src/main/java/org/matrix/androidsdk/data/Room.java]: Sometimes we use `getState()` and sometimes we use `getLiveState()`. Actually `getLiveState()` is just an alias for `getState()`. We should consider keeping only one method for clarity.

Comment (giomfo) [reply matrix-sdk/src/main/java/org/matrix/androidsdk/data/Room.java]: ok

File: matrix-sdk/src/main/java/org/matrix/androidsdk/data/Room.java (hunk 3 above)
Comment (giomfo) [reply matrix-sdk/src/main/java/org/matrix/androidsdk/data/Room.java]: ok
